R/mod_08_geo_ch_flag_drug.R

```diff
@@ -338,128 +246,111 @@ mod_08_geo_ch_flag_drug_server <- function(id, export_data) {
     # Helper functions ---------------------------------------------------------
 
     # One: spline chart
-    spline_chart_plot = function(df, df_select, fy, metric, prefix, suffix,
-                                 bottom_plot = FALSE){
-
+    spline_chart_plot <- function(df, df_select, fy, metric, plot_pos = "") {
       # Shared y axis max value across all 3 plots
-      y_axis_max_val = max(df$`20/21`, df$`21/22`, df$`22/23`)
-      accuracy = \(val) ifelse(
-        startsWith(metric, "Total") & (val < 10^3), 
-        1,
-        0.01
-      )
-      
+      y_axis_max_val <- max(df$`20/21`, df$`21/22`, df$`22/23`, df$`23/24`)
+
+      accuracy <- \(val) {
+        ifelse(startsWith(metric, "Total") & (val < 10^3), 1, 0.01)
+      }
+
       # Process original df
-      df = df %>%
+      df <- df %>%
         dplyr::rename_at(fy, ~"VALUE") %>%
         dplyr::mutate(
           VALUE_LABEL = scales::label_comma(
             accuracy = accuracy(VALUE),
-            scale_cut = scales::cut_long_scale()
+            # See this issue for reason why append of 1 is necessary:
+            # https://github.com/r-lib/scales/issues/413#issuecomment-1876179071
+            scale_cut = append(scales::cut_long_scale(), 1, 1)
           )(janitor::round_half_up(VALUE, 2))
         ) %>%
         dplyr::arrange(VALUE) %>%
         dplyr::mutate(
           index = dplyr::row_number(),
           rank = rev(dplyr::row_number()),
           total = max(rank),
-          col =  "#f7a35c",
-          label = dplyr::case_when(
-            GEOGRAPHY_CHILD == df_select ~ paste0(fy, ":   ", prefix, VALUE_LABEL, suffix),
-            TRUE ~ ""
+          col = "#f7a35c",
+          title = dplyr::case_when(
+            GEOGRAPHY_CHILD == df_select ~ glue::glue("{fy}<br>{rank} of {total}")
           )
         )
 
-      hc = highcharter::highchart() %>%
-        highcharter::hc_add_series(
-          df,
-          "spline",
-          highcharter::hcaes(index, VALUE),
-          showInLegend = FALSE,
-          dataLabels = list(
-            y = 40,
-            enabled = TRUE,
-            format = "{point.label}",
-            style = list(fontSize = "15px", fontFamily = "arial")
+      title <- df %>%
+        dplyr::select(title) %>%
+        tidyr::drop_na() %>%
+        dplyr::pull()
+
+      hc <- highcharter::highchart() %>%
+        highcharter::hc_title(
+          text = title,
+          style = list(
+            fontSize = "12px",
+            fontFamily = "arial",
+            useHTML = TRUE
           )
         ) %>%
+        highcharter::hc_add_series(df,
+          "spline",
+          highcharter::hcaes(rank, VALUE),
```

Comment:
Replaced `index` with `rank` in `hcaes`. This has flipped the charts horizontally, but I think the intention was always for this to be rank.